README.md

```diff
@@ -0,0 +1,157 @@
+# 📚Book Shop📚
+Welcome to the Online Bookstore API! This API provides essential
+functionalities for managing books, user authentication,
+and order processing, forming the backbone of a scalable and
+efficient online bookstore. Inspired by the need for a
+robust solution
+in the digital book retail space, this project showcases
+a fully functional online
+bookstore built with Spring Boot. It includes features
+like user registration, book browsing, shopping cart
+management, and order processing, offering a solid foundation
+for real-world applications.
+
+## 🛠️Technologies and Tools🛠️
+-**Spring Boot**: Core framework for building application.
+
+-**Spring Security**: Ensures robust authentication and authorization mechanisms for the application.
+
+-**Spring Data JPA**: Simplifies database interactions through JPA.
+
+-**Swagger**: Offers interactive API documentation and testing capabilities.
+
+-**JUnit**: Enables efficient unit testing.
+
+-**Mockito**: Aids in isolating code under test by mocking its dependencies, facilitating effective unit testing.
+
+-**MYSQL Database**: For the relational database.
+
+-**Liquibase**: Manages database migrations.
+
+## Features and Functionality
+### Authentication
+- **Register user**: `POST /api/auth/register`
+- **Login user**: `POST /api/auth/login`
+### Books
+- **User endpoints**
+    - **List all books**: `GET /api/books`
+    - **Get book details**: `GET /api/books/{id}`
+    - **Search book by criteria**: `GET /api/books/search?author= &title=`
+- **Admin endpoint**
+    - **Add new book**: `POST /api/books`
+    - **Update book**: `PUT /api/books/{id}`
+    - **Delete book**: `DELETE /api/books/{id}`
+### Categories
+- **User endpoints**
+    - **List all categories**: `GET /api/categories`
+    - **Get category by id**: `GET /api/categories/{id}`
+    - **Get all books by category**: `GET /api/categories/{id}/books`
+- **Admin endpoint**
+    - **Add new category**: `POST /api/categories`
+    - **Update category**: `PUT /api/categories/{id}`
+    - **Delete category**: `DELETE /api/categories/{id}`
+### Shopping Cart
+- **User endpoints**
+    - **Add book to cart**: `POST /api/cart`
+    - **Update book quantity in cart**: `PUT /api//cart-items/{cartItemId}`
+    - **Get user's cart**: `GET /api/cart`
+    - **Remove item from cart by id**: `DELETE /api//cart-items/{cartItemId}`
+### Orders
+- **User endpoints**
+    - **Get order history**: `GET /api/orders`
+    - **Place order**: `POST /api/orders`
+    - **Get all items from order**: `GET /api/orders/{id}/items`
+    - **Get specific item from order**: `GET /api/orders/{orderId}/items/{itemId}`
+- **Admin endpoint**
+    - **Update order status**: `PATCH /api/orders/{id}`
+
+## Demo video
+[📹 Watch video](https://www.loom.com/share/2e6af377787e40c98345b227dbaba7c2?sid=8e6ac58f-8de2-4e0c-b83d-f6133f6fc444)
+
+### Installation
```

Comment:
let's move this topic right after the project description, because it is quite important and usually the first thing that people look for